tests/ast/ast_test_utils.py

```diff
@@ -68,6 +72,76 @@ def generate_error_trace_info(python_text, exception=None):
     return error_msg
 
 
+def is_valid_ast_obj(ast):
+    return isinstance(ast, Iterable) or (
+        isinstance(ast, Message) and ast.ByteSize() > 0
+    )
+
+
+def get_dependent_var_ids(ast):
+    dependent_ids = set()
```

Comment:
can we add a doc string here? This function looks complex and could benefit from documentation.